src/reducers/statusEffectsTick.js

```diff
@@ -2,21 +2,35 @@ import { TICK } from '../actions/tick';
 import { BLEED_OUT } from '../actions/bleeding';
 import { FEEL_TIRED } from '../actions/tired';
 
-// This came from the status effects file. TODO: abstract this somewhere else
-const isBleeding = effects => effects.find(effect => effect.type === BLEED_OUT) || { amount: 0 };
-const isTired = effects => effects.find(effect => effect.type === FEEL_TIRED) || { amount: 0 };
+// TODO: duped from stats
+const clamp = value => Math.max(Math.min(value, 100), 0);
+
+const adjustmentsByEffect = {
+  [BLEED_OUT]: { stat: 'health', increase: false },
+  [FEEL_TIRED]: { stat: 'energy', increase: false },
+};
+
+function statsWithEffectUpdate(stats, effect) {
+  const adjustment = adjustmentsByEffect[effect.type];
+
+  // Skip effects that aren't simple stat adjustment
+  if (!adjustment) {
+    return stats;
+  }
+
+  const amount = adjustment.increase ? effect.amount : -effect.amount;
+  const updatedValue = clamp(stats[adjustment.stat] + amount);
+  return { ...stats, [adjustment.stat]: updatedValue };
+}
 
 const statusEffectsTick = (state, action) => {
   if (action.type !== TICK) {
     return state;
   }
+
   return {
     ...state,
-    stats: {
-      ...state.stats,
-      health: Math.max(state.stats.health - isBleeding(state.statusEffects).amount, 0),
-      energy: Math.max(state.stats.energy - isTired(state.statusEffects).amount, 0),
-    },
+    stats: state.statusEffects.reduce(statsWithEffectUpdate, state.stats),
```

Comment:
Reduce crash course:

Given `output = array.reduce(f, input)`, `input` and `output` should be the same sort of thing (in this case a stat block). The array can be thought of as a list of things that indicate how you want to modify your value. `f` is a function that takes a single element from the array and the current working value (i.e. something somewhere on the spectrum between `input` and `output`) and returns the updated working value based on the array element. In this case, `f` takes the effect and the current stats, and returns stats modified by the effect.

The upshot is we take a stat block and a list of effects, and return the new stat block that results from applying the changes each of those effects neccessitate.

Long-run, that should probably be a two-step process (`effects.map(adjustmentForEffect)` + `adjustments.reduce(applyAdjustment, stats)`).